### ISSUE TITLE
Add pagination to "Followed Hashtags"

### ISSUE THREAD

**Issue by zeitschlag** [mastodon/mastodon-ios#1152]
When refactoring the "Followed Hashtags" we learned, that it doesn't support pagination. So, in case, someone follows more than 100 hashtags, they'd only see the first 100.

This has been there since Mastodon `4.1`, so you'd have to distinguish between `Instance`-versions. In the end, it's probably a duplication of `FollowingViewModel` with its state machine.

**Comment by iOSampson** [comment]
Has this issue been solved? If not, I would like to give it a go.

**Comment by zeitschlag** [comment]
@SampsonEzieme Hey, thank you for that offer!

This issue hasn't been solved yet. So please, give it a try :) Do you need some hints? If you have any questions, please don't hestitate to ask.

**Comment by iOSampson** [comment]
@zeitschlag  Thank you for being so kind. I just forked the repository and followed the steps in the setup.md, however, when I try to run Mastodon in the Xcode simulator I am getting multiple missing packages and other package-related errors. Am I doing something wrong? Do you know how I should fix this?

**Comment by zeitschlag** [comment]
@SampsonEzieme Yeah, I also have issues frequently, when I change branches, for example. I blame Xcode for this. What usually helps is to update to the latest package versions fromt he menu:

<img width="631" alt="Screenshot 2024-05-24 at 22 02 07" src="https://github.com/mastodon/mastodon-ios/assets/2580019/1b070720-3340-4b7a-bb1a-b21e3fc28cab">

**Comment by iOSampson** [comment]
@zeitschlag I gave your suggestion a go, but I'm still running into errors with the package dependencies. Would you happen to have any other suggestions? Here is a screenshot of exactly what I am seeing. Again thanks for helping me out.

<img width="530" alt="Screenshot 2024-05-24 at 12 05 35 PM" src="https://github.com/mastodon/mastodon-ios/assets/63993951/33b2be2d-08c5-4112-a8ff-b4e39c527c34">

**Comment by NiwakaDev** [comment]
@SampsonEzieme 

Are you still working on this issue? If not, I'd like to work on this issue.


By the way, 
> I gave your suggestion a go, but I'm still running into errors with the package dependencies. Would you happen to have any other suggestions? Here is a screenshot of exactly what I am seeing. Again thanks for helping me out.

I followed https://github.com/mastodon/mastodon-ios/blob/develop/Documentation/Setup.md, and then I resolved the same error.

**Comment by iOSampson** [comment]
Hello @NiwakaDev

Yeah I gave it a try but I am still getting the same errors. You can go ahead and work on this issue.

**Comment by yjjem** [comment]
Hi @zeitschlag. Thank you for saving some issues for us.

Read about Tag entity and how it's used in different API Instance-versions and got some ideas how to add pagination logic.

But I'm confused about duplication viewModel and Instance-version. 
What do want to do about it?
- Do we want to combine Followed Tags and Following using v2 search.
- or just Implement distinguished pagination logic for Followed Tags ?
